app/data_client.go

```diff
@@ -0,0 +1,781 @@
+		Label:          proto.Label,
+		XMinNormalized: proto.XMinNormalized,
+		YMinNormalized: proto.YMinNormalized,
+		XMaxNormalized: proto.XMaxNormalized,
+		YMaxNormalized: proto.YMaxNormalized,
+	}
+}
+
+func annotationsFromProto(proto *pb.Annotations) Annotations {
+	if proto == nil {
+		return Annotations{}
+	}
+	bboxes := make([]BoundingBox, len(proto.Bboxes))
+	for i, bboxProto := range proto.Bboxes {
+		bboxes[i] = boundingBoxFromProto(bboxProto)
+	}
+	return Annotations{
+		Bboxes: bboxes,
+	}
+}
+
+func annotationsToProto(annotations Annotations) *pb.Annotations {
+	var protoBboxes []*pb.BoundingBox
+	for _, bbox := range annotations.Bboxes {
+		protoBboxes = append(protoBboxes, &pb.BoundingBox{
+			Id:             bbox.ID,
+			Label:          bbox.Label,
+			XMinNormalized: bbox.XMinNormalized,
+			YMinNormalized: bbox.YMinNormalized,
+			XMaxNormalized: bbox.XMaxNormalized,
+			YMaxNormalized: bbox.YMaxNormalized,
+		})
+	}
+	return &pb.Annotations{
+		Bboxes: protoBboxes,
+	}
+}
+
+func methodParamsFromProto(proto map[string]*anypb.Any) map[string]string {
+	methodParameters := make(map[string]string)
+	for key, value := range proto {
+		structValue := &structpb.Value{}
+		if err := value.UnmarshalTo(structValue); err != nil {
+			return nil
+		}
+		methodParameters[key] = structValue.String()
+	}
+	return methodParameters
+}
+
+func captureMetadataFromProto(proto *pb.CaptureMetadata) CaptureMetadata {
+	if proto == nil {
+		return CaptureMetadata{}
+	}
+	return CaptureMetadata{
+		OrganizationID:   proto.OrganizationId,
+		LocationID:       proto.LocationId,
+		RobotName:        proto.RobotName,
+		RobotID:          proto.RobotId,
+		PartName:         proto.PartName,
+		PartID:           proto.PartId,
+		ComponentType:    proto.ComponentType,
+		ComponentName:    proto.ComponentName,
+		MethodName:       proto.MethodName,
+		MethodParameters: methodParamsFromProto(proto.MethodParameters),
+		Tags:             proto.Tags,
+		MimeType:         proto.MimeType,
+	}
+}
+
+func captureMetadataToProto(metadata CaptureMetadata) *pb.CaptureMetadata {
+	methodParms, err := protoutils.ConvertStringMapToAnyPBMap(metadata.MethodParameters)
+	if err != nil {
+		return nil
+	}
+	return &pb.CaptureMetadata{
+		OrganizationId:   metadata.OrganizationID,
+		LocationId:       metadata.LocationID,
+		RobotName:        metadata.RobotName,
+		RobotId:          metadata.RobotID,
+		PartName:         metadata.PartName,
+		PartId:           metadata.PartID,
+		ComponentType:    metadata.ComponentType,
+		ComponentName:    metadata.ComponentName,
+		MethodName:       metadata.MethodName,
+		MethodParameters: methodParms,
+		Tags:             metadata.Tags,
+		MimeType:         metadata.MimeType,
+	}
+}
+
+func binaryDataFromProto(proto *pb.BinaryData) BinaryData {
+	return BinaryData{
+		Binary:   proto.Binary,
+		Metadata: binaryMetadataFromProto(proto.Metadata),
+	}
+}
+
+func binaryDataToProto(binaryData BinaryData) *pb.BinaryData {
```

Comment:
I noticed that this function is only called in the tests. I also did something similar originally, but I realized that `go-staticcheck` did not like that (see below), and I agree.
<img width="561" alt="image" src="https://github.com/user-attachments/assets/346b1085-e771-4d2e-b9d1-8d6a89a5bce0">
The purpose of the main code is to provide code functionality, while testing is supposed to test said functionality. Testing should be isolated, and writing functions that will only be used in tests is not really adding to code functionality. I'd suggest putting this function in `data_test.go`.

Reply:
This also applies to the other private functions that are only used in the test